Update diff2html to version 3

campersau opened the pull request:
just to keep up with the API changes.

See https://github.com/rtfpessoa/diff2html/pull/297#issuecomment-576030591

Hirse commented:
Are there no changes required to the CSS?

campersau commented on public/less/d2h.less:
```diff
@@ -1,218 +1,434 @@
+  background: transparent;
+}
+
+/*** adjustments for ungit ***/
```
Everything above this line is copied from https://github.com/rtfpessoa/diff2html/blob/4e28c5c3275f6f703a6923d23d001bfb676331a6/src/ui/css/diff2html.css
Everything below this line is customized for ungit.
Maybe we should split these into separate files.

Hirse commented on public/less/d2h.less:
```diff
@@ -1,218 +1,434 @@
+ *
+ * Diff to HTML (diff2html.css)
+ * Author: rtfpessoa
+ *
```
Maybe add the version number of diff2html here.

campersau replied:
Not sure, this part is copied 1:1

campersau replied:
Opted to include the css file from npm package so this file now only contains the adjustments.

campersau commented:
NEW | OLD

![image](https://user-images.githubusercontent.com/4009570/73962653-1b534300-490f-11ea-8182-be91f4f3dba0.png)

![image](https://user-images.githubusercontent.com/4009570/73962995-c06e1b80-490f-11ea-809b-27238b0400dc.png)

Hirse commented:
Thanks for the comparison images.
Where does the difference in height come from?
Also, in the bottom case (red) the line number color changes.

campersau commented:
The height difference came from the different font (monospace) and font-size (13px) which I have changed back to the previous values here https://github.com/FredrikNoren/ungit/pull/1273/commits/14a6e9be86f6f7b4e01b6c8eb9f81789b3434af0

Some other things which changed:
- The lines which contain changes are no longer indented a little bit more than the lines without changes
- The line numbers are now always aligned to the right
- The line numbers are now the same color in `line by line` vs `side by side` diffs
- Also fixed the checkbox alignment for patches

Updated screenshots from the new version:
![image](https://user-images.githubusercontent.com/4009570/74086345-48743280-4a82-11ea-82a9-17da1628e935.png)

![image](https://user-images.githubusercontent.com/4009570/74086353-55912180-4a82-11ea-8fc6-0e2e589696d1.png)

![image](https://user-images.githubusercontent.com/4009570/74086397-be789980-4a82-11ea-84cb-a8cbc6bab9b3.png)

![image](https://user-images.githubusercontent.com/4009570/74086695-a8200d00-4a85-11ea-8f87-2057cb556ae1.png)

![image](https://user-images.githubusercontent.com/4009570/74086706-c128be00-4a85-11ea-88da-55da09db11ec.png)

![image](https://user-images.githubusercontent.com/4009570/74086712-cf76da00-4a85-11ea-97ba-068a42ff6a35.png)

Hirse commented on public/less/d2h.less:
```diff
@@ -1,218 +1,64 @@
-  }
+.d2h-code-linenumber, .d2h-code-side-linenumber, .d2h-code-line-prefix {
+  -moz-user-select: none;
+  -ms-user-select: none;
```
These don't seem to be necessary any more:
https://developer.mozilla.org/en-US/docs/Web/CSS/user-select#Browser_compatibility

campersau replied:
Only in the [diff2html UI](https://github.com/rtfpessoa/diff2html#diff2htmlui-usage) which we are not using. Also it seems a bit buggy. I have opened https://github.com/rtfpessoa/diff2html/issues/303 for that.
